### PR TITLE
[368] Add delete user to support

### PR DESCRIPTION
**Pull request by stellenberger**
### Context

- [This trello ticket](https://trello.com/c/x5QrpcuK/368-allow-users-to-be-deleted-in-the-support-ui)

### Changes proposed in this pull request

- Add delete user functionality to support console

### Guidance to review

- Go to support console
- select user tab
- select a user by clicking their email address
- scroll to bottom of page and click delete this user

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by stellenberger** [comment]
[Allow users to be deleted in the support UI](https://trello.com/c/x5QrpcuK/368-allow-users-to-be-deleted-in-the-support-ui)

File: app/controllers/support/users_controller.rb [hunk 1]
```diff
@@ -21,6 +21,14 @@ def create
       end
     end
 
+    def destroy
+      if user.destroy
```
Comment (dankmitchell) [inline app/controllers/support/users_controller.rb:25]: Does `destroy` call the `discard` action? i.e. soft-delete

Comment (stellenberger) [reply app/controllers/support/users_controller.rb]: No, a hard delete; changed to `discard` for soft-delete 👍

File: spec/features/support/users/deleting_a_user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    users_show_page.delete_button.click
+  end
+
+  def then_i_am_taken_to_the_user_index_page
+    expect(users_index_page).to be_displayed
+  end
+
+  def with_a_success_message
+    expect(users_index_page).to have_content("User successfully deleted")
+  end
```
Comment (malcolmbaig) [inline spec/features/support/users/deleting_a_user_spec.rb:38]: Although feature specs typically focus on UI interactions, I think an additional step which checks the soft-deleted state of the user would be useful.

Comment (stellenberger) [reply spec/features/support/users/deleting_a_user_spec.rb:39]: good idea; added 👍

**Review by malcolmbaig — APPROVED** [review]

**Pull request merged**